File: tests/machine/x/php/join_multi.php
<?php
$customers = [
    ["id" => 1, "name" => "Alice"],
    ["id" => 2, "name" => "Bob"]
];
$orders = [
    ["id" => 100, "customerId" => 1],
    ["id" => 101, "customerId" => 2]
];
$items = [
    ["orderId" => 100, "sku" => "a"],
    ["orderId" => 101, "sku" => "b"]
];
$result = (function() use ($customers, $items, $orders) {
    $result = [];
    foreach ($orders as $o) {
        foreach ($customers as $c) {
            if ($o['customerId'] == $c['id']) {
                foreach ($items as $i) {
                    if ($o['id'] == $i['orderId']) {
                        $result[] = [
    "name" => $c['name'],
    "sku" => $i['sku']
];
                    }
                }
            }
        }
    }
    return $result;
})();
echo "--- Multi Join ---", PHP_EOL;
foreach ($result as $r) {
    var_dump($r['name'], "bought item", $r['sku']);
}
?>

File: tests/human/x/clj/join_multi.clj
(ns join-multi)

(def customers
  [{:id 1 :name "Alice"}
   {:id 2 :name "Bob"}])

(def orders
  [{:id 100 :customerId 1}
   {:id 101 :customerId 2}])

(def items
  [{:orderId 100 :sku "a"}
   {:orderId 101 :sku "b"}])

(def result
  (for [o orders
        c customers :when (= (:customerId o) (:id c))
        i items :when (= (:orderId i) (:id o))]
    {:name (:name c) :sku (:sku i)}))

(println "--- Multi Join ---")
(doseq [r result]
  (println (:name r) "bought item" (:sku r)))
